src/main/java/com/google/cloud/pontem/AvroUtil.java

```diff
@@ -0,0 +1,141 @@
+    SchemaBuilder.RecordBuilder<Schema> baseRecordBuilder =
+        SchemaBuilder.record(recordName).namespace(AVRO_SCHEMA_NAMESPACE);
+    baseRecordBuilder.prop("pontemAvroSchemaFormattingVersion", AVRO_RECORD_VERSION);
+
+    SchemaBuilder.FieldAssembler<Schema> schemaBuilderFieldAssembler = baseRecordBuilder.fields();
+
+    for (String columnName : tableInformation.getColumnNames()) {
+      schemaBuilderFieldAssembler
+          .name(columnName)
+          .type(tableInformation.getAvroTypeOfColumn(columnName))
+          .noDefault();
+    }
+
+    Schema schema = schemaBuilderFieldAssembler.endRecord();
+    LOG.info("Schema:\n" + schema.toString());
+
+    return schema;
+  }
+
+  public static Schema getAvroTypeFromSpannerType(Type cloudSpannerType, boolean isNullable) {
+    SchemaBuilder.TypeBuilder<Schema> builder = SchemaBuilder.builder();
+    switch (cloudSpannerType.getCode()) {
+      case STRING:
+        if (isNullable) {
+          return builder.nullable().stringType();
+        } else {
+          return builder.stringType();
+        }
+      case BOOL:
+        if (isNullable) {
+          return builder.nullable().booleanType();
+        } else {
+          return builder.booleanType();
+        }
+      case INT64:
+        if (isNullable) {
+          return builder.nullable().longType();
+        } else {
+          return builder.longType();
+        }
+      case FLOAT64:
+        if (isNullable) {
+          return builder.nullable().doubleType();
+        } else {
+          return builder.doubleType();
+        }
+      case BYTES:
+        if (isNullable) {
+          return builder.nullable().bytesType();
+        } else {
+          return builder.bytesType();
+        }
+      case TIMESTAMP:
+        if (isNullable) {
+          return builder.nullable().stringType();
+        } else {
+          return builder.stringType();
+        }
+      case DATE:
+        if (isNullable) {
+          return builder.nullable().stringType();
+        } else {
+          return builder.stringType();
+        }
+      case ARRAY:
+        // Types contained in arrays cannot include "NOT NULL" in syntax, so they are not nullable.
+        // I.e., DDL can only contain "ARRAY<STRING(MAX)> NOT NULL" and
+        // NOT "ARRAY<STRING(MAX) NOT NULL> NOT NULL"
+        Schema avroArrayType =
+            getAvroTypeFromSpannerType(cloudSpannerType.getArrayElementType(), false);
+        if (isNullable) {
+          return SchemaBuilder.builder().nullable().array().items().type(avroArrayType);
+        } else {
+          return SchemaBuilder.builder().array().items().type(avroArrayType);
+        }
+      default:
+        throw new IllegalArgumentException("Unknown Cloud Spanner type " + cloudSpannerType);
+    }
+  }
+
+  public static String getAvroSchemaFileLocation(String tableName) {
+    return "metadata/" + AvroUtil.AVRO_SCHEMA_FOLDER_LOCATION + "/" + tableName + ".json";
```

Comment:
Should we be using System File Separator here to address interop? Since this is using GCS is assume not but, worth the ask